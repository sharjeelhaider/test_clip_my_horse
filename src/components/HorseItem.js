import React from 'react';
import PropTypes from 'prop-types';
import {Image, StyleSheet, Text, View} from 'react-native';

const HorseItem = props => {
  const {picture, name} = props.horse;
  return (
    <View style={styles.item}>
      <Image style={styles.image} source={{uri: picture}} />
      <View style={styles.itemInner}>
        <Text style={styles.title}>{name}</Text>
        <Image
          style={styles.imageCorner}
          source={require('..//assets/icon_next.png')}
        />
        <View style={styles.divider} />
      </View>
    </View>
  );
};

HorseItem.propTypes = {
  horse: PropTypes.object.isRequired,
};

export const styles = StyleSheet.create({
  item: {
    flex: 1,
    flexDirection: 'row',
    paddingVertical: 8,
    paddingHorizontal: 8,
    marginBottom: 2,
    backgroundColor: 'black',
  },
  itemInner: {
    flex: 1,
    justifyContent: 'space-between',
  },
  divider: {
    backgroundColor: 'grey',
    height: 2,
    marginTop: 2,
  },
  image: {
    width: 150,
    height: 100,
  },
  imageCorner: {
    width: 40,
    height: 40,
    alignSelf: 'flex-end',
  },
  title: {
    fontSize: 16,
    marginLeft: 10,
    fontWeight: 'bold',
    color: 'white',
    paddingBottom: 4,
  },
});

export default HorseItem;
