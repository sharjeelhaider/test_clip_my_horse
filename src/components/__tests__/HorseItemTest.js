import React from 'react';
import HorseItem from '../HorseItem';
import mockedHorses from '../../../config/jest/dataMock';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders a HorseItem using Snapshots', () => {
  const wrapper = renderer.create(<HorseItem horse={mockedHorses[0]} />);
  expect(wrapper).toMatchSnapshot();
});
