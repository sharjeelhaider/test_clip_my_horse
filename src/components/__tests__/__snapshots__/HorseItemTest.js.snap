// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a HorseItem using Snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 2,
      "paddingHorizontal": 8,
      "paddingVertical": 8,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://d2vtwgzbrv2xnd.cloudfront.net/51e87826c943fd896385734287bbac27fd43c74a.jpeg",
      }
    }
    style={
      Object {
        "height": 100,
        "width": 150,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginLeft": 10,
          "paddingBottom": 4,
        }
      }
    >
      American Beauty
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icon_next.png",
        }
      }
      style={
        Object {
          "alignSelf": "flex-end",
          "height": 40,
          "width": 40,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "grey",
          "height": 2,
          "marginTop": 2,
        }
      }
    />
  </View>
</View>
`;
